File: leetcode/001-two-sum/TwoSum.java
class Solution {
    public int[] twoSum(int[] nums, int target) {
        Map<Integer, Integer> numbersPositions = new HashMap<>();
        for (int i = 0; i < nums.length; i++) {
            if (numbersPositions.containsKey(target - nums[i])) {
                return new int[]{ numbersPositions.get(target - nums[i]), i };
            }
            
            numbersPositions.put(nums[i], i);
        }
        
        return new int[]{ -1, -1 };
    }
}

File: leetcode/001-two-sum/TwoSum.swift
//
//  TwoSum.swift
//  LeetCode
//
//  Created by Valerii Iatsko on 10/7/20.
//

import Foundation

class TwoSum {
    func twoSum(_ nums: [Int], _ target: Int) -> [Int] {
        var dict: [Int: Int] = [:]

        for i in 0..<nums.count {
            if let seenIndex = dict[target - nums[i]] {
                return [seenIndex, i]
            }

            dict[nums[i]] = i
        }

        return [-1, -1]
    }
}
